### PR TITLE
fix(install): Don't respect MSRV for non-local installs 

### PR DESCRIPTION
**Pull request by epage**
### What does this PR try to resolve?

This is part of #9930

### CODE REVIEW

File: src/cargo/core/workspace.rs [hunk 1]
```diff
@@ -103,7 +103,7 @@ pub struct Workspace<'gctx> {
 
     /// The resolver behavior specified with the `resolver` field.
     resolve_behavior: ResolveBehavior,
-    resolve_honors_rust_version: bool,
+    resolve_honors_rust_version: Option<bool>,
```
Comment (weihanglo) [inline src/cargo/core/workspace.rs:106]: Wonder why we need an `Option` here. It doesn't seem too relevant to this PR.

File: src/cargo/core/workspace.rs [hunk 2]
```diff
@@ -657,10 +657,15 @@ impl<'gctx> Workspace<'gctx> {
         self.honor_rust_version
     }
 
+    pub fn set_resolve_honors_rust_version(&mut self, honor_rust_version: Option<bool>) {
+        self.resolve_honors_rust_version = honor_rust_version;
+    }
+
     pub fn resolve_honors_rust_version(&self) -> bool {
```
Comment (weihanglo) [inline src/cargo/core/workspace.rs:664]: I am now seeing `resolve_honors_rust_version` and `honors_rust_version` quiet confusing. Any change to explain or conslidate them?

File: src/cargo/core/workspace.rs (hunk 1 above)
Comment (epage) [reply src/cargo/core/workspace.rs:106]: Oops, that represents a bug.  Fixing it.

File: src/cargo/core/workspace.rs (hunk 2 above)
Comment (weihanglo) [reply src/cargo/core/workspace.rs:664]: BTW, not a blocker

File: src/cargo/core/workspace.rs (hunk 1 above)
Comment (epage) [reply src/cargo/core/workspace.rs:106]: I misread, there wasn't a bug and I didn't need to take advantage of this.  Simplifying the code.

**Review by weihanglo — APPROVED** [review]
Thanks!

**Pull request merged**